libp2p/stream/bufferstream.nim

```diff
@@ -129,14 +129,16 @@ proc initBufferStream*(s: BufferStream,
   if not(isNil(handler)):
     s.writeHandler = proc (data: seq[byte]) {.async, gcsafe.} =
       defer:
-        s.writeLock.release()
+        if s.writeLock.locked:
```

Comment:
why do you need to check that the lock is acquired? what sequence causes it not to be acquired?

Reply:
If the lock isn't acquired `release` will throw - maybe not needed, I'm either way on this one.

Reply:
well, it should be well understood at this point whether the lock is locked or not - and whether it has purpose at all? "maybe" locked locks seems like reason to be worried.

Reply:
Yeah I agree, this wont happen if the log is acquired at the right time and it doesn't happen here or anywhere, it was by chance that I hit it at some point and decided that ti would be a good check overall, but now that I think about it, this check can actually be detrimental, because it would hide issues with lock acquisitions.